pkg/apis/core/validation/validation.go

```diff
@@ -5531,6 +5485,68 @@ func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod, opts PodVali
 	return allErrs
 }
 
+// ValidatePodResize tests that a user update to pod container resources is valid.
+// newPod and oldPod must only differ in their Containers[*].Resources and
+// Containers[*].ResizePolicy field.
+func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) field.ErrorList {
+	// Part 1: Validate newPod's spec and updates to metadata
+	fldPath := field.NewPath("metadata")
+	allErrs := ValidateObjectMetaUpdate(&newPod.ObjectMeta, &oldPod.ObjectMeta, fldPath)
```

Comment:
Should we change this to be:
```suggestion
	allErrs := ValidateImmutableField(&newPod.ObjectMeta, &oldPod.ObjectMeta, fldPath)
```

Reply:
I think validation should not assume anything about the calling context.

Reply:
> ValidateImmutableField naming would be incorrect as this function is being used in pod updates too.

What do you mean? Isn't this function (`ValidatePodResize`) only called for the resize subresource strategy?

> I think validation should not assume anything about the calling context.

Is that an argument for not checking metadata at all, not assuming that metadata changes have been reset?

Reply:
To clarify my original comment: there's already a function `ValidateImmutableField`, I was just suggesting we call that one instead.

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L360-L362

Reply:
> Is that an argument for not checking metadata at all, not assuming that metadata changes have been reset?

The latter - if we are validating that only X and Y are changed, then that should include asserting that metadata has not changed.

Reply:
I initially thought you meant renaming `ValidateObjectMetaUpdate` to `ValidateImmutableField`. But I realized now that you meant calling an existing func `ValidateImmutableField`. I added this block again.